File: common/src/main/java/de/nicidienase/chaosflix/common/entities/streaming/Group.java
package de.nicidienase.chaosflix.common.entities.streaming;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by felix on 23.03.17.
 */

public class Group {
	String group;
	List<Room> rooms;

	public Group() {}

	public Group(String group) {
		this.group = group;
	}

	public String getGroup() {
		return group;
	}

	public void setGroup(String group) {
		this.group = group;
	}

	public List<Room> getRooms() {
		return rooms;
	}

	public void setRooms(List<Room> rooms) {
		this.rooms = rooms;
	}

	public static Group getDummyObject(){
		Group dummy = new Group();
		dummy.setGroup("Dummy Group");
		dummy.setRooms(new ArrayList<>());
		dummy.getRooms().add(Room.getDummyObject());
		return dummy;
	}
}

File: common/src/main/java/de/nicidienase/chaosflix/common/entities/streaming/Group.kt
package de.nicidienase.chaosflix.common.entities.streaming

import android.arch.persistence.room.Entity

import java.util.ArrayList

@Entity(tableName = "group")
data class Group(
    var group: String,
    var rooms: MutableList<Room>){


    companion object {
        val dummyObject: Group
            get() {
                val dummy = Group("Dummy Group", ArrayList<Room>())
                dummy.rooms.add(Room.dummyObject)
                return dummy
            }
    }
}
